src/Symfony/Component/DependencyInjection/Loader/Configurator/Traits/ArgumentTrait.php

```diff
@@ -22,6 +30,25 @@
      */
     final public function args(array $arguments)
     {
+        $arguments = array_map(
```

Comment:
Ihe inner can be injected anywhere, not only in the constructor, and can be nested also. The logic should be in AbstractConfigurator::processValue() (but it's not trivial to do so).

Reply:
> and can be nested also

What do you mean?

> The logic should be in AbstractConfigurator::processValue() (but it's not trivial to do so)

It is still in dev branch, later will be much harder to fix it. Probably, it's not the only case when we need `Definition` object there.

Reply:
the inner service doesn't have to be injected as an argument: it can be nested in an array, which itself is passed as argument (constructor, or setter, or property in fact.)

Reply:
> the inner service doesn't have to be injected as an argument: it can be nested in an array, which itself is passed as argument (constructor, or setter, or property in fact.)

It just requires to tweak `AbstractConfigurator::processValue()` to make it work, doesn't it?

Reply:
it should (but I'm still not sure it's worth the added complexity)

Reply:
It there any reason why it is static? As I can see, `AbstractConfigurator` contains link to the `definition`. I'm not big fan of it, but it's cheap solution.

Reply:
it's used in `function iterator(array $values)`